scanreport.sh

```diff
@@ -1,21 +1,24 @@
 #!/bin/bash
 
 function usage {
-	echo "usage: $1 [-f nmap.grepable] [-i IP] [-p port] [-s service] [-P protocol]"
+	echo "Usage: $1 [-f nmap.grepable] [-i IP] [-p port] [-s service] [-P protocol] [-a]"
+	echo "  -a: outputs matching IP addresses only"
 }
 
 db=""
 ip=""
 port=""
 all=0
 proto=""
-while getopts "f:i:p:P:s:" OPT; do
+addressonly=""
+while getopts "f:i:p:P:s:a" OPT; do
 	case $OPT in
 		f) db=$OPTARG;;
 		i) ip=$OPTARG;;
 		p) port=$OPTARG;;
 		s) service=$OPTARG;;
 		P) proto=$OPTARG;;
+		a) addressonly=true;;
```

Comment:
Can we use this instead to keep it consistent since you're using $2 later on as the IP address:

`addressonly=$OPTARG;;`